### PR TITLE
[WIP] Come back to haproxy

### PR DESCRIPTION
**Pull request by aojea**
Seems that haproxy is the only load balancer that allows to check the health endpoint from the api servers while doing TCP load balancing.

This PR restores Haproxy as the load balancer for the kind cluster, and leverages the docker commands `create` and `start` to create the load balancer node.

Fixes: #588

### CODE REVIEW

**Comment by aojea** [comment]
/assign @neolit123 
/assign @BenTheElder 

I still need to do more testing, but the simple scenario is working in my laptop

File: pkg/container/docker/run.go [hunk 1]
```diff
@@ -62,7 +62,7 @@ func WithMounts(mounts []cri.Mount) RunOpt {
 }
 
 // Run creates a container with "docker run", with some error handling
-func Run(image string, opts ...RunOpt) error {
+func Run(image string, createonly bool, opts ...RunOpt) error {
```
Comment (BenTheElder) [inline pkg/container/docker/run.go]: -1 too boolean args, this is not `docker.Run` anymore

File: pkg/cluster/internal/loadbalancer/const.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ package loadbalancer
 const ControlPlanePort = 6443
 
 // Image defines the loadbalancer image:tag
-const Image = "nginx:1.15.12-alpine"
+const Image = "haproxy:1.9.8-alpine"
```
Comment (BenTheElder) [inline pkg/cluster/internal/loadbalancer/const.go]: we should go back to the previous approach of using our own image with a default config. I have this WIP.

**Comment by aojea** [comment]
Duplicates

**Pull request closed without merging**